total-value-and-liquidity/src/agent.js

```diff
@@ -51,63 +48,71 @@ async function parseData(dataPromise, reserve) {
   return parsedData;
 }
 
-function provideHandleBlock(RollingMath, config, lendingPool, dataProvider) {
-  rollingLiquidityData = {};
-
+function provideHandleBlock(data) {
   return async function handleBlock(blockEvent) {
+    const {
+      lendingPoolContract,
+      dataProviderContract,
+      dataFields,
+      rollingLiquidityData,
+      windowSize,
+      numStdDeviations,
+      minElements,
+    } = data;
     const findings = [];
 
     // override block number so we get data from the block in question
     const override = { blockTag: blockEvent.blockNumber };
 
     // get reserves and the current liquidity from this block
-    const reserves = await lendingPool.getReservesList({ ...override });
+    const reserves = await lendingPoolContract.getReservesList({ ...override });
 
     // create array containing a promise that returns the data for each reserve
-    const reserveData = [];
+    const reserveDataPromises = [];
     reserves.forEach((reserve) => {
-      // RPC call for per reserve data
-      const dataPromise = dataProvider.getReserveData(reserve, { ...override });
-      reserveData.push(
-        parseData(dataPromise, reserve).catch(() => undefined),
+      // RPC call per reserve data
+      const dataPromise = dataProviderContract.getReserveData(reserve, { ...override });
+      reserveDataPromises.push(
+        parseData(dataPromise, reserve, dataFields).catch(() => undefined),
       );
     });
 
     // resolve our requests
-    const resolved = (await Promise.all(reserveData)).filter(
+    const resolved = (await Promise.all(reserveDataPromises)).filter(
       (result) => result !== undefined,
     );
```

Comment:
Functionally, this should be the same, but I think it's more readable:
```suggestion
    const reserveDataPromises = reserves.map(async (reserve) => {
      try {
        // RPC call per reserve data
        const promise = dataProviderContract.getReserveData(reserve, { ...override });
        return parseData(promise, reserve, dataFields);
      } catch (error) {
        console.error(error);
        return [];
      }
    });
    
    const resolved = (await Promise.all(reserveDataPromises)).flat();
```